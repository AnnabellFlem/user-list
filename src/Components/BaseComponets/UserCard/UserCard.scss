@import 'src/Styles/sizes';
@import 'src/Styles/colors';
@import 'src/Styles/media';

.user {
  margin: $indent-m 0;
  border-radius: $border-radius-s;
  width: calc(33.3% - 2 * #{$indent-m});
  min-width: 280px;
  overflow: hidden;
  box-shadow: 0 4px 8px $black-5, 0 0 0 1px $white-10 inset;
  position: relative;

  @include tablet {
    width: calc(50% - 2 * #{$indent-m});
  }

  @include mobile {
    width: calc(100% - 2 * #{$indent-s});
    margin: $indent-s;
  }

  &__avatar {
    width: 50px;
    height: 50px;
    border-radius: 50%;
    overflow: hidden;
    font-size: 0;
    margin-right: $indent-xs;
  }

  &__header {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    padding: $indent-m;
    background: linear-gradient(
      to bottom,
      rgba(0, 0, 0, 0.05) 0%,
      rgba(0, 0, 0, 0.05) 1%,
      rgba(0, 0, 0, 0) 100%
    );
  }

  &__data {
    flex: 1;
    white-space: nowrap;
    text-align: right;
    overflow: hidden;
    text-overflow: ellipsis;
    padding-left: $indent-m;
  }

  &__info {
    padding: $indent-s $indent-m;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
    font-size: $font-xs;
    line-height: $lh-s;
  }

  &__info + &__info {
    border-top: 1px solid $white-50;
  }

  &__title {
    font-size: $font-xs;
    font-weight: bold;
    color: $black-40;
    letter-spacing: 1px;
  }

  &__btn--edit {
    border: 1px solid $black-5;
    background-color: $white-10;
    border-radius: $border-radius-s;
    padding: $indent-xxs $indent-xs;
    color: $black-100;
    font-size: $font-xs;
    margin: $indent-s $indent-m;
    transition: all 0.2s ease;

    &:hover {
      box-shadow: 0 5px 10px $black-5;
    }
  }
}
