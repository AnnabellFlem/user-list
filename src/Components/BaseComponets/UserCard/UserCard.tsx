import React from 'react'
import './UserCard.scss'

type UserCardProp = {
  user: {
    id: number
    name: string
    surname: string
    email: string
    phone: string
    birthday: string
  }
}

const getGender = (userId: number) => (userId % 2 === 0 ? 'men' : 'women')

const UserCard: React.FC<UserCardProp> = ({ user }) => {
  return (
    <li className="user" id={`user-${user.id}`}>
      <div className="user__header">
        <div className="user__avatar">
          <img
            src={`https://randomuser.me/api/portraits/${getGender(user.id)}/${
              user.id
            }.jpg`}
          />
        </div>
      </div>
      <div className="user__info">
        <span className="user__title">Name</span>
        <span className="user__data">{user.name}</span>
      </div>
      <div className="user__info">
        <span className="user__title">Surname</span>
        <span className="user__data">{user.surname}</span>
      </div>
      <div className="user__info">
        <span className="user__title">Email</span>
        <span className="user__data">{user.email}</span>
      </div>
      <div className="user__info">
        <span className="user__title">Phone</span>
        <span className="user__data">{user.phone}</span>
      </div>
      <div className="user__info">
        <span className="user__title">Birthday</span>
        <span className="user__data">{user.birthday}</span>
      </div>
    </li>
  )
}

export default UserCard
