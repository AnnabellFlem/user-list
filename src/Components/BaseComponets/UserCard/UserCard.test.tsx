import React from 'react'
import { shallow } from 'enzyme'
import UserCard from './UserCard'

const mockedData = {
  id: 'id',
  name: 'name',
  surname: 'surname',
  email: 'email',
  phone: 'phone',
  birthday: 'birthday',
  credit: 'credit',
}

const mockedHandleClick = jest.fn()

describe('UserCard test', () => {
  let wrapper: any

  beforeAll(() => {
    wrapper = shallow(
      <UserCard user={mockedData} handleClick={mockedHandleClick} />,
    )
  })

  it('should to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Component should render li', () => {
    expect(wrapper.find('li')).toHaveLength(1)
  })

  it('Component should render user info', () => {
    expect(wrapper.find('.user__data').at(0).text()).toBe(mockedData.name)
    expect(wrapper.find('.user__data').at(1).text()).toBe(mockedData.surname)
  })

  it('Component should render button', () => {
    expect(wrapper.find('button')).toHaveLength(1)
  })
})
