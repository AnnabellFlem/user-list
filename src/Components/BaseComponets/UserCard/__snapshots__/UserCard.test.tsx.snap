// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard test should to match snapshot 1`] = `ShallowWrapper {}`;
